### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by Shivani-1524**
Added login and signup functionality

files to check:
AuthPages folder
Navbar.js

### CODE REVIEW

File: src/Pages/AuthPages/LoginPage.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const navigate = useNavigate()
+    const [loginData, setLoginData] = useState({ email: '', password: '' })
+    const [loginError, setLoginError] = useState(false);
+    const [userNotFoundError, setUserNotFoundError] = useState(false);
+
+    const handleUserLogin = async (e) => {
+        e.preventDefault()
+        try {
+            const res = await axios.post('/api/auth/login', { ...loginData })
+            console.log(res)
```
Comment (Naman-Saxena1) [inline src/Pages/AuthPages/LoginPage.js]: Can remove this console

File: src/Pages/AuthPages/LoginPage.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                            props={{ labelFor: 'password', labelTitle: 'Password', placeholderText: "Enter test", objKey: 'password', inputType: 'password' }} />
+
+                        {/* <div className="mg-t-20">
+                            <label className="input-label sm-title" htmlFor="name">Email</label>
+                            <input className="user-input" required onChange={(e) => setLoginData(prev => ({ ...prev, email: e.target.value }))} id="name" type="email" />
+                        </div> */}
+                        {/* <div className="mg-t-10">
+                            <label className="input-label sm-title" htmlFor="password">Password</label>
+                            <input className="user-input" required onChange={(e) => setLoginData(prev => ({ ...prev, password: e.target.value }))} id="password" type="password" placeholder="Enter test" />
+                        </div> */}
```
Comment (Naman-Saxena1) [inline src/Pages/AuthPages/LoginPage.js]: If not needed, can remove commented code

File: src/Pages/AuthPages/SignupPage.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }
+
+    // const formFieldData = [
+    //     { labelFor: 'firstName', labelTitle: 'First Name', placeholderText: 'Enter First Name', objKey: 'firstName' },
+    //     { labelFor: 'lastName', labelTitle: 'Last Name', placeholderText: 'Enter Last Name', objKey: 'lastName' },
+    //     { labelFor: 'email', labelTitle: 'Email', placeholderText: 'Enter email address', objKey: 'email' },
+    //     { labelFor: 'password', labelTitle: 'Password', placeholderText: "6+ characters", objKey: 'password' }
+    // ]
+
```
Comment (Naman-Saxena1) [inline src/Pages/AuthPages/SignupPage.js:44]: Same here, 2 consoles in handleUserSignup and commented code.

**Review by Naman-Saxena1 — CHANGES_REQUESTED** [review]
Only the home page is opening, not sure if it is intentional or a bug.
You cross check if you are able to open other pages in Netlify preview.

File: src/App.js [hunk 4]
```diff
@@ -1,12 +1,18 @@
 import './App.css';
 import { HomePage } from './Pages/HomePage/HomePage';
+import LoginPage from './Pages/AuthPages/LoginPage';
+import LogoutPage from './Pages/AuthPages/LogoutPage';
+import SignupPage from './Pages/AuthPages/SignupPage';
```
Comment (yaswanthmyneni) [inline src/App.js]: suggestion: use index.js folder in pages and export all pages from it

File: src/App.js [hunk 5]
```diff
@@ -1,12 +1,18 @@
...
 import { BrowserRouter, Routes, Route } from 'react-router-dom';
 
 function App() {
   return (
     <BrowserRouter>
       <div className="App">
```
Comment (yaswanthmyneni) [inline src/App.js]: changes: remove < BrowserRouter > from App.js

File: src/Components/Navbar/Navbar.js [hunk 6]
```diff
@@ -1,40 +1,51 @@
...
+        <nav className="navbar">
+            <div className="nav-logo">
                 <img src={NavLogo} alt="logo" />
             </div>
-            <div class="nav-items">
-                <Link to="/" class="nav-link sm-title hide-md" href="./">About Us</Link>
-                <a class="nav-link sm-title hide-md" href="./Products/products.html">Shop</a>
+            <div className="nav-items">
+                <Link to="/" className="nav-link sm-title hide-md" href="./">About Us</Link>
+                <a className="nav-link sm-title hide-md" href="./Products/products.html">Shop</a>
```
Comment (yaswanthmyneni) [inline src/Components/Navbar/Navbar.js:18]: changes: remove href='./ from < Link > tag

File: src/Pages/AuthPages/SignupPage.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    });
+    const handleUserSignup = async (e) => {
+        e.preventDefault()
+        try {
+            const res = await axios.post(
+                '/api/auth/signup',
+                { ...signupData }
+            )
+            console.log(res)
+            if (res.status === 200 || res.status === 201) {
```
Comment (yaswanthmyneni) [inline src/Pages/AuthPages/SignupPage.js:22]: changes: remove console.log() when you push code to github

**Review by yaswanthmyneni — APPROVED** [review]
you can merge after changes.

File: src/App.js (hunk 4 above)
Comment (yaswanthmyneni) [reply src/App.js]: and also use either named export or default export. In the above code, HomePage is named export and LoginPages is the default export.

File: src/Pages/AuthPages/SignupPage.js (hunk 7 above)
Comment (Shivani-1524) [reply src/Pages/AuthPages/SignupPage.js:22]: done

File: src/App.js (hunk 4 above)
Comment (Shivani-1524) [reply src/App.js]: done. i will just use named exports from now on.

**Comment by Shivani-1524** [comment]
> Only the home page is opening, not sure if it is intentional or a bug. You cross check if you are able to open other pages in Netlify preview.

Did you try clicking on the user icon in the navbar ? the login page can be previewed there.

**Pull request merged**